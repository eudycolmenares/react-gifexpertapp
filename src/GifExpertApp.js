import { useState } from 'react';
import CategoryAdd from './components/CategoryAdd';
import GifGrid from './components/GifGrid'
//
const GifExpertApp = ({ defaultCategories = [] }) => {
    const [categories, setCategories] = useState( defaultCategories )
    return (
        <>
            <h2>GifExpertApp</h2>
            <CategoryAdd setCategories={ setCategories } />
            <hr />
            <ol>
                {
                    categories.map(category => 
                        <GifGrid 
                            category={ category }
                            key={category} 
                        />
                    )
                }
            </ol>
        </>
    )
}
// 
export default GifExpertApp