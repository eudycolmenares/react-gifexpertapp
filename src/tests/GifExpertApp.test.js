import { shallow } from 'enzyme'
import GifExpertApp from '../GifExpertApp'
// 
describe('Tests in <GifExpertApp />', () => {
    test('Should display correctly!', () => {
        const wrapper = shallow( <GifExpertApp /> )
        expect( wrapper ).toMatchSnapshot()
    })
    test('Should display a list of categories!', () => {
        const categories = ['Pokemon', 'Digimon']
        const wrapper = shallow( <GifExpertApp defaultCategories={categories} /> )
        expect( wrapper ).toMatchSnapshot()
        expect( wrapper.find('GifGrid').length ).toBe( categories.length )
    })
})