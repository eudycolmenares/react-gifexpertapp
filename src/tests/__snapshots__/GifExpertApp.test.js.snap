// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifExpertApp /> Should display a list of categories! 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Pokemon"
      key="Pokemon"
    />
    <GifGrid
      category="Digimon"
      key="Digimon"
    />
  </ol>
</Fragment>
`;

exports[`Tests in <GifExpertApp /> Should display correctly! 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
